pennylane/grouping/group_observables.py

```diff
@@ -0,0 +1,258 @@
+    optimal partitioning, i.e. the fewest number of cliques, is the minimum clique cover (MCC)
+    problem. The solution of MCC may be found by graph colouring on the complementary graph. Both
+    MCC and graph colouring are NP-Hard, so heuristic graph colouring algorithms are employed to
+    find approximate solutions in polynomial time.
+
+    Args:
+        observables (list[Observable]): A list of Pauli words to be partitioned according to a
+        grouping strategy.
+
+    Keyword Args:
+        grouping_type (str): The binary relation used to define partitions of the Pauli words.
+        graph_colourer (str): The heuristic algorithm to employ for graph colouring.
+
+    Raises:
+        ValueError: if `grouping_type` or `graph_colourer` are not recognized as elements of
+        `GROUPING_TYPES` or `GRAPH_COLOURING_METHODS` respectively.
+
+    """
+
+    def __init__(self, observables, grouping_type="qwc", graph_colourer="rlf"):
+
+        if grouping_type.lower() not in GROUPING_TYPES:
+            raise ValueError(
+                "Grouping type must be one of: {}, instead got {}.".format(
+                    GROUPING_TYPES, grouping_type
+                )
+            )
+
+        self.grouping_type = grouping_type.lower()
+
+        if graph_colourer.lower() not in GRAPH_COLOURING_METHODS.keys():
+            raise ValueError(
+                "Graph colouring method must be one of: {}, instead got {}.".format(
+                    list(GRAPH_COLOURING_METHODS.keys()), graph_colourer
+                )
+            )
+
+        self.graph_colourer = GRAPH_COLOURING_METHODS[graph_colourer.lower()]
+        self.observables = observables
+        self.__wire_map = None
+        self.__n_qubits = None
+        self.binary_observables = None
+        self.adj_matrix = None
+        self.grouped_paulis = None
+
+    def obtain_binary_repr(self, n_qubits=None, wire_map=None):
+        """Converts the list of Pauli words to a binary matrix.
+
+        Keyword args:
+            n_qubits (int): number of qubits to specify dimension of binary vector representation.
+            wire_map (dict): dictionary containing all wire labels used in the Pauli word as keys,
+                and unique integer labels as their values
+
+        Returns:
+            array[bool]: a column matrix of the Pauli words in binary vector representation.
+
+        """
+
+        if wire_map is None:
+            self.__wire_map = {
+                wire: c
+                for c, wire in enumerate(
+                    Wires.all_wires([obs.wires for obs in self.observables]).tolist()
+                )
+            }
+
+        else:
+            self.__wire_map = wire_map
+
+        self.binary_observables = convert_observables_to_binary_matrix(
+            self.observables, n_qubits, self.__wire_map
+        )
+
+        self.__n_qubits = n_qubits
+
+        return self.binary_observables
```

Comment:
Might be worth separating the call to `convert_observables_to_binary_matrix` from setting the `self.binary_observables` attribute. The reason for that is that the `obtain_binary_repr` method indicates that one will obtain the binary representation of the observables. However, it is does not imply (fairly enough) that the `self.binary_observables` attribute will also be set, making the setting of this attribute a side effect.

(This has an implication for later usage, leaving a comment there too.)
```suggestion

        self.__n_qubits = n_qubits

        return convert_observables_to_binary_matrix(self.observables, n_qubits, self.__wire_map)
```